backend/hydroshare_jupyter_sync/resource_hydroshare_data.py

```diff
@@ -0,0 +1,334 @@
+                "sizeBytes": #,
+                "type": "folder",
+                "contents": [list of similar dicts representing files and folders],
+            }
+         """
+        if self._files and not force_fetch:
+            return self._files
+
+        hs_resource_info = self.hs.getResourceFileList(self.res_id)
+        url_prefix = 'http://www.hydroshare.org/resource/' + self.res_id + '/data/contents'
+        folders_dict = {}
+        root_dir_contents = []
+        nested_files = {}
+        for file_info in hs_resource_info:
+            # extract filepath from url
+            filepath = file_info["url"][len(url_prefix)+1:]
+            # get proper definition formatting of file if it is a file
+            file_definition_hs = self.get_file_metadata(filepath, filepath, file_info, HS_PREFIX+':')
+            # if it is a folder, build up contents
+            if not file_definition_hs:
+                nested_files[filepath + "/"] = file_info
+                folders = filepath.split("/")
+                currpath = ""
+                for x in range(0, len(folders)-1):
+                    folder = folders[x]
+                    currpath = currpath + folder + "/"
+                    # build up dictionary of folder -> what is in it
+                    if (x, folder, currpath) not in folders_dict:
+                        folders_dict[(x, folder, currpath)] = []
+                    folders_dict[(x, folder, currpath)].append((x+1, folders[x+1], currpath + folders[x+1] + "/"))
+            # if it is just a file, add it to the final list
+            else:
+                root_dir_contents.append(file_definition_hs)
+
+        # go through folders dictionary & build up the nested structure
+        i = 0
+        for key, val in folders_dict.items():
+            # we only want to make the initial call on folders at the top level,
+            # (level 0); folders at levels 1, 2, etc. will be built into the
+            # result by means of the recursive calls
+            if key[0] == 0:
+                folder_time, folder_size, folder_contents = self.get_contents_recursive(val, folders_dict,
+                                                                                         nested_files, HS_PREFIX+':')
+                if folder_time:
+                    folder_time = str(folder_time)
+                # TODO (vicky): use name and path instead of key[i]
+                root_dir_contents.append({
+                    "name": key[1],
+                    "path": HS_PREFIX + ':/' + key[2].strip('/'),
+                    "sizeBytes": folder_size,
+                    "modifiedTime": folder_time,
+                    "type": "folder",
+                    "contents": folder_contents,
+                })
+
+        # TODO: probably add some comments explaining what these lines are doing
+        rootsize = 0
+        for f in root_dir_contents:
+            rootsize += (f["sizeBytes"])
+
+        self._files = {
```

Comment:
Did we not have this calling resource_last_modified_time for the upper level "/" folder's modified time before? Any reason we are removing that? Right now, I am not sure how the frontend is accessing resource_last_modified_time function without this.

Reply:
I don't think I touched this function much... If it was getting the last modified time for a resource, I removed it because the frontend gets that information when it gets a list of all the resources.

Reply:
so does this mean we don't need the get_resource_last_modified function then? Is it ever being used?

Reply:
Oh huh you're right, we're not using it anymore. I'm definitely getting the right modified date though, so it's getting extracted from the metadata somewhere else. 
![image](https://user-images.githubusercontent.com/3915596/79693530-d150dc80-8230-11ea-9832-e159a19ec5ad.png)

Reply:
Also, amazingly, I haven't been able to break that resource yet. (\*\*knocks on wood\*\*)

Reply:
yup, that's totally fine, you are probably just pulling it from the resource metadata, which means we can likely remove that function

Reply:
OK, removed.